.circleci/config.yml

```diff
@@ -1,75 +1,129 @@
+           echo 'Building Electron for release'
+           script/build.py -c R
+         else
+           echo 'Building Electron for debug'
+           script/build.py -c D
+         fi
+    - run:
+        name: Create distribution
+        command: |
+          if [ "$ELECTRON_RELEASE" == "1" ]; then
+            echo 'Creating Electron release distribution'
+            script/create-dist.py
+          else
+            echo 'Skipping create distribution because build is not for release'
+          fi
+    - run:
+        name: Upload distribution
+        command: |
+          if [ "$ELECTRON_RELEASE" == "1" ] && [ "$TRIGGERED_BY_API" != "1" ]; then
+            echo 'Uploading Electron release distribution to github releases'
+            script/upload.py
+          elif [ "$ELECTRON_RELEASE" == "1" ] && [ "$TRIGGERED_BY_API" == "1" ]; then
+            echo 'Uploading Electron release distribution to s3'
+            script/upload.py --upload_to_s3
+          else
+            echo 'Skipping upload distribution because build is not for release'
+          fi
+    - run:
+        name: Setup for headless testing
+        command: |
+          if [ "$ELECTRON_RELEASE" != "1" ] && [ "$RUN_HEADLESS_TESTS" == "true" ]; then
+            echo 'Setup for headless testing'
+            sh -e /etc/init.d/xvfb start
+          else
+            echo 'Headless testing not needed'
+          fi
+    - run:
+        name: Test
+        environment:
+          MOCHA_FILE: junit/test-results.xml
+          MOCHA_REPORTER: mocha-junit-reporter
+        command: |
+          if [ "$ELECTRON_RELEASE" != "1" ] && [ "$RUN_TESTS" == "true" ]; then
+            echo 'Testing Electron debug build'
+            mkdir junit
+            script/test.py --ci
+          else
+            if [ "$ELECTRON_RELEASE" == "1" ]; then
+              echo 'Skipping testing on release build'
+            else
+              echo 'Skipping tests due to configuration'
+            fi
+          fi
+    - run:
+        name: Verify FFmpeg
+        command: |
+          if [ "$ELECTRON_RELEASE" != "1" ] && [ "$RUN_TESTS" == "true" ]; then
+            echo 'Verifying ffmpeg on debug build'
+            script/verify-ffmpeg.py
+          else
+            if [ "$ELECTRON_RELEASE" == "1" ]; then
+              echo 'Skipping verify ffmpeg on release build'
+            else
+              echo 'Skipping tests due to configuration'
+            fi
+          fi
+    - run:
+        name: Generate Typescript Definitions
+        command: |
+          if [ "$CREATE_TYPESCRIPT_DEFS" == "true" ]; then
+            npm run create-typescript-definitions
+          fi
+    - persist_to_workspace:
+        root: /home/builduser
+        paths:
+          - project/out          
+    - store_test_results:
+        path: junit
+    - store_artifacts:
+        path: junit
+    - store_artifacts:
+        path: out/electron.d.ts
+    - store_artifacts:
+        path: out/electron-api.json
```

Comment:
Nice job removing the (mostly?) duplicated rules :+1:

Reply:
Yes.  I used https://circleci.com/blog/circleci-hacks-reuse-yaml-in-your-circleci-config-with-yaml/ as a guide to deduplicate.